src/common/snippets/src/lowered/pass/cleanup_loop_offsets.cpp

```diff
@@ -22,54 +22,60 @@ bool CleanupLoopOffsets::run(LinearIR& linear_ir) {
     const auto before_last = std::prev(linear_ir.end());
     for (auto expr_it = linear_ir.begin(); expr_it != before_last; expr_it++) {
         const auto& node = expr_it->get()->get_node();
-        if (auto loop_end = as_type_ptr<op::LoopEnd>(node)) {
-                auto next_expr_it = std::next(expr_it);
-                const auto& next_node = next_expr_it->get()->get_node();
-                // Note: Finalization offsets before the Result can be safely disregarded
-                // TODO: Need verify that Buffers on the inputs doesn't have other consumers (other Loops)
-                //       and this Loop doesn't have Buffer on other outputs.
-                if (is_type<ov::op::v0::Result>(next_node)) {
-                    const auto& fin_offsets = loop_end->get_finalization_offsets();
-                    loop_end->set_finalization_offsets(std::vector<int64_t>(fin_offsets.size(), 0));
-                    is_modified = true;
-                }
-                if (auto outer_loop_end = as_type_ptr<op::LoopEnd>(next_node)) {
-                    auto fin_offsets = loop_end->get_finalization_offsets();
-                    std::unordered_map<PortConnectorPtr, size_t> per_port_connector_offset;
-                    const auto& loop_inputs = expr_it->get()->get_input_port_connectors();
-                    for (size_t i = 0; i < fin_offsets.size(); i++)
-                        per_port_connector_offset[loop_inputs[i]] = i;
+        if (auto loop_end = as_type_ptr<op::LoopEndStatic>(node)) {
+            auto next_expr_it = std::next(expr_it);
```

Comment:
Removed redundant tab. There are only two changes: added `if (!outer_is_incremented[i])` and introduced `LoopEndStatic`